packages/strapi-design-system/src/DatePicker/DatePickerTd.tsx

```diff
@@ -1,14 +1,13 @@
-import React from 'react';
+import { MouseEventHandler } from 'react';
 
-import PropTypes from 'prop-types';
-import styled from 'styled-components';
+import styled, { ThemeColors } from 'styled-components';
 
-import { RawTd } from '../RawTable';
+import { RawTd, RawTdProps } from '../RawTable';
 import { Typography } from '../Typography';
 
-const DatePickerCellButton = styled.button`
+const DatePickerCellButton = styled.button<{ $isSelected?: boolean }>`
   border: none;
-  background: ${({ theme, isSelected }) => (isSelected ? theme.colors.primary100 : theme.colors.neutral0)};
+  background: ${({ theme, $isSelected }) => ($isSelected ? theme.colors.primary100 : theme.colors.neutral0)};
```

Comment:
That's interesting cause the docs recommend the dollar sign – https://styled-components.com/docs/api#transient-props

Reply:
Keep this then + create a follow-up ticket to get rid of the transient prop stuff?

Reply:
I think that would a breaking change on many components 🤔 e.g. `Box` it might have to wait for a v2 thing. I can still write an issue and add a label to signify it's breaking though?